### PR TITLE
Added ruleset for structs vs classes

### PR DESCRIPTION
**Pull request by Haud**
@ThibaultKlein @paulmiard

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -172,6 +173,19 @@ class Object {
...
+
+In Swift, structs maintain value semantics which means their values are copied when assigned. Classes, on the other hand, act like pointers from C
+and Objective-C; they are called reference types and the internal data is shared amongst instances of assigning.
+
+When composing your types, consider carefully what they're going to be used for before choosing what they should end up being. In general, 
+types that are immutable, stateless, and can have a definition for equality should be `struct`; for cases where an object is stateful, mutable,
+or where equality does not make sense, favor `class`.
+
+In general, you should favor structs and protocols over classes; even in cases where polymorphism would dictate the usage of a class, consider if you can
+achieve a similar result via protocols and extensions.
```
Comment (ThibaultKlein) [inline README.md]: if we could mention the keyword `composition` versus `inheritance` it would make me happy ;)

**Comment by ThibaultKlein** [comment]
it would be good to have bullet points to make it easier to read, currently I feel that the information is hidden in the paragraphs, we should make it very easy to find information in the README

**Comment by Haud** [comment]
@ThibaultKlein updated based on your feedback. I moved the basic points to bullet points, added some more tangible benefits for struct-usage, and mentioned composition and inheritance JUST FOR YOU !

**Comment by Haud** [comment]
Also paging @paulmiard

File: README.md [hunk 2]
```diff
@@ -172,6 +173,28 @@ class Object {
...
+
+In Swift, structs maintain value semantics which means their values are copied when assigned. Classes, on the other hand, act like pointers from C
+and Objective-C; they are called reference types and the internal data is shared amongst instances of assigning.
+
+When composing your types, consider carefully what they're going to be used for before choosing what they should end up being. In general, 
+consider structs for types that are:
+
+* immutable
+* stateless
+* have a definition for equality
```
Comment (ThibaultKlein) [inline README.md]: Capitalize?

**Comment by ThibaultKlein** [comment]
:+1:

File: README.md (hunk 2 above)
Comment (Haud) [inline README.md]: done

**Pull request merged**